### ISSUE TITLE
Versioning for subflows

### ISSUE THREAD

**Issue by zackwasli** [FlowFuse/flowfuse#3230]
### Description

The ability to create snapshots for specific subflows, so that updating a subflow can be automatic across varying instances of Node-RED.

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

**Comment by joepavitt** [comment]
Summary of Request:

- Have a subflow defined from a Node-RED instance
- Save that flow to centralised library
- Use it in other instances
- Update the centralised library entry
- Auto update to all instances using that entry

**Comment by zackwasli** [comment]
@joepavitt yes this is correct. This slack thread was referencing this issue, but I have no further details at this time: **https://flowforgeworkspace.slack.com/archives/C03FAUBPPJ7/p1715201725090749?thread_ts=1715004346.071099&cid=C03FAUBPPJ7**
